src/OrchardCore.Modules/Orchard.Liquid/Filters/MetadataFilters.cs

```diff
@@ -1,63 +1,56 @@
 using System;
-using DotLiquid;
-using Microsoft.AspNetCore.Html;
+using Fluid;
+using Fluid.Values;
 using Microsoft.AspNetCore.Mvc;
-using Microsoft.Extensions.DependencyInjection;
 using Orchard.ContentManagement;
-using Orchard.Liquid.Drops;
 
 namespace Orchard.Liquid.Filters
 {
-    public static class MetadataFilters
+    public class MetadataFilters : ITemplateContextHandler
     {
-        public static string DisplayText(Context context, object input)
+        private readonly IContentManager _contentManager;
+
+        public MetadataFilters(IContentManager contentManager)
         {
-            var contentItem = (input as ContentItemDrop)?.ContentItem;
+            _contentManager = contentManager;
+        }
 
-            if (contentItem == null)
+        public void OnTemplateProcessing(TemplateContext context)
+        {
+            context.Filters.AddFilter("display_text", (input, arguments, ctx) =>
             {
-                throw new ArgumentException("Content Item was expected");
-            }
+                var contentItem = input.ToObjectValue() as ContentItem;
```

Comment:
input is a `FluidValue` it wraps any CLR object. So yes we want to get the wrapped content item then cast it.